scripts/update_hook_revs.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Script to ensure the pre-commit hook revs match what is installed."""
+import re
+import sys
+from functools import partial
+from pathlib import Path
+
+from script_utils import cli
+
+REPO_ROOT_DIR = Path(__file__).parent.parent.resolve()
+PRE_COMMIT_CFG_PATH = REPO_ROOT_DIR / ".pre-commit-config.yaml"
+LOCK_FILE_PATH = REPO_ROOT_DIR / "requirements-dev.txt"
+
+
+def inspect_lock() -> dict[str, str]:
+    """Inspect the lock file to get the resolved dependencies"""
+    dependency_pattern = re.compile(r"([^=\s]+)==([^\s]*?)\s")
+    dependencies = {}
+
+    # Get the set of dependencies from the requirements-dev.txt lock file
+    with open(LOCK_FILE_PATH, encoding="utf-8") as lock_file:
+        lines = lock_file.readlines()
+
+    for line in lines:
+        match = re.match(dependency_pattern, line)
+        if match:
+            package, version = match.groups()
+            dependencies[package] = version
+
+    return dependencies
+
+
+def get_repl_value(match, dependencies: dict[str, str], outdated_hooks: list[str]):
+    """Look up pre-commit hook id in list of dependencies. If there's a match, update
+    return the version stored in the dictionary"""
```

Comment:
Something is wrong with the grammar of the docstring.

Reply:
Does it update? Does it return? Who knows? 
Edited it in the commit below.